wiki-service/src/main/java/org/exoplatform/wiki/mow/core/api/wiki/PageImpl.java

```diff
@@ -601,13 +602,27 @@ public synchronized String addRelatedPage(PageImpl page) throws Exception {
       while (refferedIter.hasNext()) {
         Entry<String, Value> entry = refferedIter.next();
         PageImpl page = chSession.findById(PageImpl.class, entry.getValue().getString());
-        if(page != null && page.hasPermission(PermissionType.VIEWPAGE)){
-          relatedPages.add(page);
+        if(!isRelatedPageRemoved(page)){

```

Comment:
relation should be deleted in the relation deletion logic, but check when getting relation list.